### PR TITLE
Calculate stats - Prompts Sent vs Data Redactions for given time period

### PR DESCRIPTION
**Pull request by shubham-k3**
Calculate stats - Prompts Sent vs Data Redactions for given time period

### CODE REVIEW

File: chats/dao/ChatHistoryDao.py [hunk 1]
```diff
@@ -239,3 +239,83 @@ def importChatHistory(self, newChatId, userId, sourceChatHistory):
                 if message.get("isDeleted", False) == False
             ]
         })
+
+    def calculateStat(self, period):
+        return self.collection.aggregate([
+            {
+                "$match": {
+                    "messages.type": "USER_INPUT",
+                    "messages.createdAt": period['createdAt'],
```
Comment (nikvin15) [inline chats/dao/ChatHistoryDao.py:248]: Is this expected ? Is it a string or object ?

File: chats/dao/ChatHistoryDao.py [hunk 2]
```diff
@@ -239,3 +239,83 @@ def importChatHistory(self, newChatId, userId, sourceChatHistory):
...
+                }
+            }
+        ])
+    
+    def calculateTopRedactedSensitiveData(self, period):
+        return self.collection.aggregate([
+            {
+                "$match": {
+                    "messages.type": "USER_INPUT",
+                    "messages.createdAt": period['createdAt'],
```
Comment (nikvin15) [inline chats/dao/ChatHistoryDao.py:295]: Same as above

File: chats/views/chatview.py [hunk 3]
```diff
@@ -216,3 +216,16 @@ def handleLlmModels(request, modelId = None):
...
+    {
+        Permission.ReadEmployeeChats: ['GET']
+    },
+    ['GET']
+)
+def calculateStats(request):
+    isUI = getTheIsUIFlag(request)
+
+    if request.method == 'GET':
+        return calculateStat(isUI)
```
Comment (nikvin15) [inline chats/views/chatview.py]: nitpick: Add new line

File: chats/views/chatview.py [hunk 4]
```diff
@@ -216,3 +216,16 @@ def handleLlmModels(request, modelId = None):
 
     if request.method == 'PUT':
         return updateLlmModel(data, modelId, isUI)
+
+
+@view_with_auth_permissions(
+    {
+        Permission.ReadEmployeeChats: ['GET']
```
Comment (nikvin15) [inline chats/views/chatview.py:223]: This would be under security admin roles

File: stats/StatsDao.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+from utils.MongoConnection import MongoConnection
+from bson.objectid import ObjectId
+from bson.json_util import dumps
+import json
+from chats.data_firewall.DataFirewall import DataFirewall
+
+TYPE_OF_AI = {
+    'GPT': 'GPT',
+    'LLAMA': 'LLAMA',
+}
```
Comment (nikvin15) [inline stats/StatsDao.py]: We have already defined this somewhere in the project. Let's move that in the utils and use at both the places

File: stats/StatsDao.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+            stat = json.loads(dumps(chatHistoryDao.calculateStat(period)))
+            stats[period['createdAt']['$gte'].strftime('%d %b') ] = stat[0] if stat else {'_id': 0, 'totalMessageCount': 0, 'totalPiiCount': 0}
+        return stats
+
+    def calculateStatsForLastNMonths(self, no_of_months = 2):
+        periods = self.getPeriodForLastNMonthsMonthly(no_of_months)
+        stats = dict()
+        for period in periods:
+            stat = json.loads(dumps(chatHistoryDao.calculateStat(period)))
+            stats[str(period['createdAt']['$gte'].strftime('%d %b')) + ' to ' + str(period['createdAt']['$lt'].strftime('%d %b'))] = stat[0] if stat else {'_id': 0, 'totalMessageCount': 0, 'totalPiiCount': 0}
```
Comment (nikvin15) [inline stats/StatsDao.py]: Can we split this into multiple lines to make it readable

File: chats/dao/ChatHistoryDao.py (hunk 1 above)
Comment (shubham-k3) [reply chats/dao/ChatHistoryDao.py:248]: object

**Review by nikvin15 — APPROVED** [review]

**Pull request merged**